### DISCUSSION TITLE
segfault when running tests

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#24918] [Q&A Getting Started] [answered]
Trying to run tests on HPC, I get (paraphrasing here...)
```
> ./run_tests --help
Building and linking hit...
Building hit for python with python3-config
make[1]: Entering directory '..../framework/contrib/hit'
make[1]: 'hit' is up to date
make[1]: Leaving directory '..../framework/contrib/hit'
Segmentation fault
```
Maybe it's one of the imports that the test harness uses that's causing the segfault? Looks like it doesn't even get around to processing the arguments.

**Comment by GiudGiud** [comment]
Hello

Is this a new copy of moose?
and what modules are you running?

Guillaume

**Reply by jessecarterMOOSE** [reply]
Sorry, should have given more details. Fresh install of Moose, running the framework tests. Didn't build any modules yet.

**Reply by GiudGiud** [reply]
I meant HPC modules

**Reply by GiudGiud** [reply]
did you at least load `use.moose moose-dev` ? (or some other good enough set)

**Reply by jessecarterMOOSE** [reply]
Oh, I'm running on our local HPC. Built petsc, libmesh, and moose fine.

**Reply by GiudGiud** [reply]
ah ok. 
Does MOOSE run fine on a simple test?

**Reply by milljm** [reply]
Because you built libMesh, can you post your `libmesh_diagnostic.log`? That contains a ton of helpful information about your working environment.

**Reply by jessecarterMOOSE** [reply]
@GiudGiud yes I can run MOOSE with a simple input problem (simple_diffusion)

@milljm no I'm on an airgapped cluster :frowning_face: what kind of info are you looking for?

**Reply by milljm** [reply]
I ask for the diagnostic logs when I myself do not know what I am looking for

They do usually illuminate environment concerns, paths to binaries with versions in them (`openmpi-4.3.1/bin` "oh I see you are using OpenMPI, version 4.3.1"). Paths to things along with modules loaded which may interfere (you have Python-3.10 loaded as a module, yet a path to python reveals you are still using `/usr/bin/python`, etc). Paths to compilers mixed with paths to other compilers (GCC _and_ Intel compilers possibly colliding for example).

Honestly, lots of things...

**Reply by jessecarterMOOSE** [reply]
I'll comb through it to see if anything jumps out. Given that PETSc/libmesh/MOOSE all get built and run fine, my instinct is that it has something to do with my python setup since it crashes on loading the TestHarness.

**Reply by jessecarterMOOSE** [reply, marked as answer]
ok I think I got this figured out. I was using the system-installed conda environment. I made my own minimal conda environment and it worked.
